### PR TITLE
Cleaning up public interface

### PR DESCRIPTION
**Pull request by NachoEmbrace**
* Adds missing objc names for some classes/structs so they now use the "EMB" prefix.
* Updated file headers (copyright 2023 > 2024)
* Added missing code level docs for some classes.
* Removed some unused files.

### CODE REVIEW

File: Sources/EmbraceCore/Capture/Network/URLSessionCaptureService+Options.swift [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-//  Copyright © 2023 Embrace Mobile, Inc. All rights reserved.
+//  Copyright © 2024 Embrace Mobile, Inc. All rights reserved.
 //
 
 import Foundation
 
 extension URLSessionCaptureService {
     /// Class used to setup a URLSessionCaptureService.
-    @objc public final class Options: NSObject {
+    @objc(URLSessionCaptureServiceOptions)
```
Comment (ArielDemarco) [inline Sources/EmbraceCore/Capture/Network/URLSessionCaptureService+Options.swift:9]: Missing the `EMB` prefix
```suggestion
    @objc(EMBURLSessionCaptureServiceOptions)
```

**Review by ArielDemarco — APPROVED** [review]

File: Sources/EmbraceCore/Capture/UX/Tap/TapCaptureService.swift [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
+//  Copyright © 2024 Embrace Mobile, Inc. All rights reserved.
 //
 #if canImport(UIKit) && !os(watchOS)
 import UIKit
 import EmbraceCaptureService
 import EmbraceCommonInternal
 import EmbraceOTelInternal
 
-/// Service that capture taps on the screen.
+/// Service that generates OpenTelemtry span events for taps on the screen.
```
Comment (NachoEmbrace) [inline Sources/EmbraceCore/Capture/UX/Tap/TapCaptureService.swift:10]:
```suggestion
/// Service that generates OpenTelemetry span events for taps on the screen.
```

File: Sources/EmbraceCore/Public/PushNotifications/PushNotificationEvent.swift [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 //
 
 import EmbraceOTelInternal
 import Foundation
 import UserNotifications
 
-public struct PushNotificationEvent: SpanEvent {
+/// Class used to represent a Push Notification as a SpanEvent.
+/// Usage example:
+/// `Embrace.client?.add(.push(userInfo: apsDictionary))
```
Comment (NachoEmbrace) [inline Sources/EmbraceCore/Public/PushNotifications/PushNotificationEvent.swift:11]:
```suggestion
/// `Embrace.client?.add(.push(userInfo: apsDictionary))`
```

File: Sources/EmbraceCore/Public/Events/Breadcrumb.swift [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
 //
 
 import Foundation
 import EmbraceOTelInternal
 import EmbraceCommonInternal
 
-public struct Breadcrumb: SpanEvent {
+/// Class used to represent a Breadcrum as a SpanEvent.
+/// Usage example:
+/// `Embrace.client?.add(.breadcrumb("This is a breadcrumb"))
```
Comment (NachoEmbrace) [inline Sources/EmbraceCore/Public/Events/Breadcrumb.swift:11]:
```suggestion
/// `Embrace.client?.add(.breadcrumb("This is a breadcrumb"))`
```

**Pull request merged**